### PR TITLE
Add integration test to check type error display

### LINKED ISSUE #347: Add integration tests that fail type checking
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

We had a panic on all programs that didn't type check (#345), and this merged into master. We should catch such problems in automated tests.

### PR DESCRIPTION
**Pull request by iwahbe**
This test is unfortunately and necessarily fragile, since it tests what the user is displayed instead of a structured result.

The test will not pass before https://github.com/pulumi/pulumi/pull/10918 merges.

Fixes #347

### CODE REVIEW

**Comment by iwahbe** [comment]
https://github.com/pulumi/pulumi/pull/10918 has merged, so we can merge this PR after the next release (which should be tomorrow).

File: pkg/tests/integration_test.go
Comment (dixler) [inline pkg/tests/integration_test.go:61]: on first glance, I'd suggest asserting that

> length: Cannot assign type 'string' to type 'integer'

and

>   lower: Cannot assign type 'number' to type 'boolean'

are in the output since this seems fragile if we change the output and those errors are controlled by this repository.

Comment (iwahbe) [reply pkg/tests/integration_test.go]: We can do that.

File: pkg/tests/integration_test.go
Comment (dixler) [inline pkg/tests/integration_test.go:25]: nit. I'd break these out into 3 different `assert.Contains` since I'm not sure if the ordering is guaranteed or if we want the test to guarantee the ordering.

**Review by dixler — APPROVED** [review]
LGTM other than a nit around possible future test flakiness, but I trust your judgement.

File: pkg/tests/integration_test.go
Comment (iwahbe) [reply pkg/tests/integration_test.go:25]: The ordering is guaranteed and important. I'm happy to have this test scream at us if the ordering changes.

Remember, the invalid part of the test file looks like this:
```yaml
  randomString:
     type: random:RandomString
     properties:
       length: "three"
       lower: 7
```
We want that `properties` (`{length: string, lower: number}`) is invalid first, and then, indented and in order, the reasons why the main property assignment is invalid.

**Comment by iwahbe** [comment]
Blocked on https://github.com/pulumi/pulumi/pull/10965.

**Review by AaronFriel — APPROVED** [review]
LGTM

**Pull request merged**
